CityOfDetroit/bloom
# Implement backend pagination for GET /listings

## Issue

Addresses #56 

## Description

This change changes the parameters and return type for `GET /listings` such that results are paginated. The parameters now include `page` and `limit` fields, and the returned object includes both an array of Listings and metadata about e.g. which page of listings this represents, how many total listings there are, how many listings per page, etc.

The pagination implementation uses `skip` and `take` when building the SQL query, and those correspond to SQL's `OFFSET` and `LIMIT`, respectively. `typeorm` actually sends two queries as a result: a first to ask for all the IDs of listings that are in the specified range, and a second to get all of the data (including all of the data in `JOIN`ed tables). (And, this change adds a third `SELECT COUNT(*) FROM listings;` query in order to populate the pagination metadata, but that should be fast.) The hope is that, although this change means more SQL queries to the database, the overall time to retrieve listing data is faster since we don't have to retrieve all of the `JOIN`ed data in a single query.

This change also adds a couple tests and fixes a couple places in the frontend so that they understand the new paginated-listings return type.

## Type of change


## How Can This Be Tested/Reviewed?

```
$ cd backend/core
$ yarn test:e2e:local
$ yarn test listing
```

Also, if you run the code locally and point your browser to `localhost:3100/listings`, you should be able to see the new paginated listings result.

## Checklist:
